lib/TPP/MapToBatchReduceGEMM.cpp

```diff
@@ -59,6 +85,45 @@ static bool isOutputOperand(linalg::LinalgOp linalgOp, OpOperand &operand) {
   return !isInputOperand(linalgOp, operand);
 }
 
+static LogicalResult checkVNNIAccessPatterns(linalg::LinalgOp linalgOp) {
+  SmallVector<AffineMap> maps;
+  for (OpOperand &operand : linalgOp->getOpOperands()) {
```

Comment:
this nesting still looks a bit entangled can you first scan the input and then the output?